packages/core/strapi/lib/commands/utils/pkg.js

```diff
@@ -0,0 +1,420 @@
+'use strict';
+
+const fs = require('fs/promises');
+const chalk = require('chalk');
+const pkgUp = require('pkg-up');
+const yup = require('yup');
+
+/**
+ * Utility functions for loading and validating package.json
+ * this includes the specific validation of specific parts of
+ * the package.json.
+ */
+
+/**
+ * The schema for the package.json that we expect,
+ * currently pretty loose.
+ */
+const packageJsonSchema = yup.object({
+  name: yup.string().required(),
+  version: yup.string().required(),
+  type: yup.string().matches(/(commonjs|module)/),
+  license: yup.string(),
+  bin: yup.mixed().oneOf([
+    yup.string(),
+    yup.object({
+      [yup.string()]: yup.string(),
+    }),
+  ]),
+  main: yup.string(),
+  module: yup.string(),
+  source: yup.string(),
+  types: yup.string(),
+  exports: yup.lazy((value) =>
+    yup.object(
+      typeof value === 'object'
+        ? Object.entries(value).reduce((acc, [key, value]) => {
+            if (typeof value === 'object') {
+              acc[key] = yup
+                .object({
+                  types: yup.string(),
+                  source: yup.string(),
+                  import: yup.string(),
+                  require: yup.string(),
+                  default: yup.string(),
+                })
+                .noUnknown(true);
+            } else {
+              acc[key] = yup
+                .string()
+                .matches(/^\.\/.*\.json$/)
+                .required();
+            }
+
+            return acc;
+          }, {})
+        : undefined
+    )
+  ),
+  files: yup.array(yup.string()),
+  scripts: yup.object(),
+  dependencies: yup.object(),
+  devDependencies: yup.object(),
+  peerDependencies: yup.object(),
```

Comment:
Probably for later: should this validate e.g. `react`, `react-dom` ... are peer dependencies?

Reply:
We have a task for doing `plugin:verify` which would include this sort of thing 👍🏼